shared/AppInsightsCommon/src/StorageHelperFuncs.ts

```diff
@@ -88,142 +190,145 @@ export function utlEnableStorage() {
  * @param reset - Should the usage be reset and determined only based on whether LocalStorage is available
  */
 export function utlCanUseLocalStorage(reset?: boolean): boolean {
-    if (reset || _canUseLocalStorage === undefined) {
-        _canUseLocalStorage = !!_getVerifiedStorageObject(StorageType.LocalStorage);
+    if (reset || !_verifiedLocalStorage) {
+        _verifiedLocalStorage = createCachedValue(_getVerifiedStorageObject(StorageType.LocalStorage));
     }
 
-    return _canUseLocalStorage;
+    return !!(_verifiedLocalStorage && _verifiedLocalStorage.v);
 }
 
 export function utlGetLocalStorage(logger: IDiagnosticLogger, name: string): string {
-    const storage = _getLocalStorageObject();
-    if (storage !== null) {
-        try {
+    try {
+        const storage = _getLocalStorageObject();
+        if (storage !== null) {
             return storage.getItem(name);
-        } catch (e) {
-            _canUseLocalStorage = false;
-
-            _throwInternal(logger,
-                eLoggingSeverity.WARNING,
-                _eInternalMessageId.BrowserCannotReadLocalStorage,
-                "Browser failed read of local storage. " + getExceptionName(e),
-                { exception: dumpObj(e) });
         }
+    } catch (e) {
+        _throwInternal(
```

Comment:
this looks like dead code as is because _getLocalStorage will never throw an exception